symfony/plugins/orangehrmLeavePlugin/Service/LeaveAssignmentService.php

```diff
@@ -200,47 +204,56 @@ public function getLeaveRequestStatus($isWeekend, $isHoliday, $leaveDate, LeaveP
         }
 
         if (is_null($status)) {
-            
             $workFlowItem = $this->getWorkflowItemForAssignAction($leaveAssignmentData);
-            if (!is_null($workFlowItem)) {
-                $status = Leave::getLeaveStatusForText($workFlowItem->getResultingState());
-            } else {                
-                throw new LeaveAllocationServiceException('Not Allowed to Assign Leave to Selected Employee!');
-            }                        
-            
-            if (($status == Leave::LEAVE_STATUS_LEAVE_APPROVED) && (strtotime($leaveDate) < strtotime(date('Y-m-d')))) {
-                $status = Leave::LEAVE_STATUS_LEAVE_TAKEN;
+            $status = Leave::LEAVE_STATUS_LEAVE_APPROVED;
+            if ($workFlowItem instanceof WorkflowStateMachine) {
+                /** @var LeaveStatus|null $leaveStatus */
+                $leaveStatus = $this->getRepository(LeaveStatus::class)->findOneBy(
+                    ['name' => $workFlowItem->getResultingState()]
+                );
+                if ($leaveStatus instanceof LeaveStatus) {
+                    $status = $leaveStatus->getStatus();
+                }
```

Comment:
can you try the below code instead of this, please refactor this and `LeaveApplicationService`

```php
class LeaveApplicationService extends AbstractLeaveAllocationService
{
    use LeaveRequestServiceTrait; // <-- make sure to use this

    // ...
    if ($workFlowItem instanceof WorkflowStateMachine) {
        $status = $this->getLeaveRequestService()->getLeaveStatusByName($workFlowItem->getResultingState());
   }
```

Reply:
done